File: msgraph-mail/go/utilities/models/entity.go
package models

import (
    i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55 "github.com/microsoft/kiota/abstractions/go/serialization"
)

type Entity struct {
    additionalData map[string]interface{};
    id *string;
}
func NewEntity()(*Entity) {
    m := &Entity{
    }
    m.setAdditionalData(make(map[string]interface{}));
    return m
}
func (m *Entity) getAdditionalData()(map[string]interface{}) {
    return m.additionalData
}
func (m *Entity) getId()(*string) {
    return m.id
}
func (m *Entity) GetFieldDeserializers()(map[string]func(interface{}, i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.ParseNode)(error), error) {
    return nil, nil
}
func (m *Entity) Serialize(writer i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.SerializationWriter)(error) {
    return nil
}
func (m *Entity) setAdditionalData(value map[string]interface{})() {
    m.additionalData = value
}
func (m *Entity) setId(value *string)() {
    m.id = value
}

File: msgraph-mail/typescript/src/models/entity.ts
import { deserializeIntoAttachment } from './attachment';
import { deserializeIntoExtension } from './extension';
import { deserializeIntoInferenceClassification } from './inferenceClassification';
import { deserializeIntoInferenceClassificationOverride } from './inferenceClassificationOverride';
import { deserializeIntoMailFolder } from './mailFolder';
import { deserializeIntoMessage } from './message';
import { deserializeIntoMessageRule } from './messageRule';
import { deserializeIntoMultiValueLegacyExtendedProperty } from './multiValueLegacyExtendedProperty';
import { deserializeIntoOutlookItem } from './outlookItem';
import { deserializeIntoSingleValueLegacyExtendedProperty } from './singleValueLegacyExtendedProperty';
import { type AdditionalDataHolder, type Parsable, type ParseNode, type SerializationWriter } from '@microsoft/kiota-abstractions';

export function createEntityFromDiscriminatorValue(parseNode: ParseNode | undefined) {
    if(!parseNode) throw new Error("parseNode cannot be undefined");
    return deserializeIntoEntity;
}
export function deserializeIntoEntity(entity: Entity | undefined = {} as Entity) : Record<string, (node: ParseNode) => void> {
    return {
        "id": n => { entity.id = n.getStringValue(); },
    }
}
export interface Entity extends AdditionalDataHolder, Parsable {
    /**
     * Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     */
    additionalData?: Record<string, unknown>;
    /**
     * The unique identifier for an entity. Read-only.
     */
    id?: string | undefined;
}
export function serializeEntity(writer: SerializationWriter, entity: Entity | undefined = {} as Entity) : void {
        writer.writeStringValue("id", entity.id);
        writer.writeAdditionalData(entity.additionalData);
}
